### DISCUSSION TITLE
Where did org.jgroups.stack.IpAddressUUID go?

### DISCUSSION THREAD

**Discussion by armlesshobo** [belaban/JGroups#660] [General]
I noticed that it exists in the latest 4.x, but was removed in 5.x. I'm curious about where that class went and what the reasoning was for its removal.

Thank you in advance :)

**Comment by tristantarrant** [comment]
This is the Jira https://issues.redhat.com/browse/JGRP-2465
and the commit that removed it is 76636ce6121f20a29c676361a07bdf82983909ad

**Comment by armlesshobo** [comment]
Thank you for your prompt response and helpful information.

What's curious is that I tried to upgrade from version 4.2.20.Final of jgroups to 5.1.6.Final and I get this error message in the log:

> 13 Sep 2022 11:03:39,445 ERROR [management-operation  Controller Boot Thread] WFLYCTL0013: Operation ("add") failed - address: ([
>     ("subsystem" => "infinispan"),
>     ("cache-container" => "web"),
>     ("thread-pool" => "transport")
> ]) - failure description: {"WFLYCTL0080: Failed services" => {"org.wildfly.clustering.infinispan.cache-container-configuration.web" => "java.lang.NoClassDefFoundError: org/jgroups/stack/IpAddressUUID
>     Caused by: java.lang.NoClassDefFoundError: org/jgroups/stack/IpAddressUUID
>     Caused by: java.lang.ClassNotFoundException: org.jgroups.stack.IpAddressUUID from [Module \"org.wildfly.clustering.server\" version 26.1.1.Final from local module loader @353efdbf (finder: local module finder @55cff952 (roots: D:\\deploy\\jboss\\modules,D:\\deploy\\jboss\\modules\\system\\layers\\base))]"}}

I think this means that the Infinispan subsystem relies on that class, still. I checked Infinispan's release candidate for version 14 and see that it uses 5.2.4 of jgroups, so I must wait for its official release before I incorporate that into WildFly. Moreover, WildFly 27 Alpha still uses version 13 of Infinispan.

Anyway, thank you again for your help!

**Reply by rhusar** [reply]
@armlesshobo No, you can't just drop in a new major version jar into a WildFly server and expect things to work. There are extensive code changes required - https://github.com/wildfly/wildfly/compare/main...pferraro:wildfly:infinispan-14.0.x

There's a branch that you can check out and build if you want to preview new jgroups and infinispan upgrades: https://github.com/pferraro/wildfly/tree/infinispan-14.0.x

**Reply by rhusar** [reply]
@armlesshobo Good news – WildFly `main` branch is not using latest JGroups.

**Comment by rhusar** [comment]
@armlesshobo WildFly 27.x is now released with JGroupx 5.x! https://www.wildfly.org/news/2022/11/09/WildFly27-Final-Released/
